src/games/marooned.rs

```diff
@@ -688,7 +690,9 @@ impl GameState {
                 target: action.remove,
             });
         }
-        Ok(self.history.push(action))
+        let mut new_game = self.clone();
```

Comment:
Is this the correct place to clone the game? `im` exposes a "mutable" api, so I could push this clone into the Elixir wrapper if I wanted

Reply:
I wouldn't necessarily expect a `clone` at all, but instead a new game. How is `history.push_back` working? Shouldn't it return a Vector?

Reply:
`history.push_back`  "mutates" the vector in place. In the case that no one else has a copy of the vector this mutates it in place. In the case where someone else is holding shared refs into the vector, it will do the cloning under the hood and change the reference that this is using (similar to the type of api we talked about wrapping in python)

Reply:
I think I'd like the signature of `make_move` to be `&self` and not `&mut self`. The only way to accomplish this is to do the cloning explicitly myself. I give up the potential optimization of the caller being the only holder into the data so it can be safely mutated without issue

Reply:
Here is the method https://docs.rs/im/15.0.0/im/vector/struct.Vector.html#method.push_back, notice it takes a `&mut self`

Reply:
Looks like there are various operations where the vector isn't actually a rbb at small sizes, it's just a regular ol vec
https://docs.rs/im/15.0.0/src/im/vector/mod.rs.html#904-915

Reply:
It's worth pointing out that `clone` is super cheap in most cases I believe https://docs.rs/im/15.0.0/src/im/vector/mod.rs.html#1690-1703